### PR TITLE
Added BuildSecrets to build manifest

### LINKED ISSUE #2566: Support for build secrets
**Is your feature request related to a problem? Please describe.**
Build secrets are now part of the compose spec:
https://github.com/compose-spec/compose-spec/pull/238/files

### PR DESCRIPTION
**Pull request by teresaromero**
# Proposed changes

Fixes #2566 

Secrets was supported via `okteto build` flag, now secrets can be added to the okteto manifest under the build section

### CODE REVIEW

**Comment by AgustinRamiroDiaz** [comment]
Do we currently support this secrets on the docker compose manifest?

File: pkg/model/serializer.go [hunk 1]
```diff
@@ -46,6 +46,7 @@ type buildInfoRaw struct {
 	VolumesToInclude []StackVolume  `yaml:"-"`
 	ExportCache      string         `yaml:"export_cache,omitempty"`
 	DependsOn        BuildDependsOn `yaml:"depends_on,omitempty"`
+	Secrets          BuildSecrets   `yaml:"secrets,omitempty"`
```
Comment (AgustinRamiroDiaz) [inline pkg/model/serializer.go:49]: Can we add a comment to this struct like

"it's an auxiliary struct that should match `BuildInfo` and it's used to handle recursion in custom un/marshalling"

So we understand why this struct has to change when `BuildInfo` changes

File: pkg/cmd/build/build.go [hunk 2]
```diff
@@ -271,6 +271,16 @@ func OptsFromBuildInfo(manifestName, svcName string, b *model.BuildInfo, o *type
 		}
 	}
 
+	if len(b.Secrets) > 0 {
+		if o.Secrets == nil {
```
Comment (AgustinRamiroDiaz) [inline pkg/cmd/build/build.go:275]: nit: this is not necessary as go handles `append` for nil slices (not the same for nil maps though)

File: pkg/cmd/build/build.go [hunk 3]
```diff
@@ -271,6 +271,16 @@ func OptsFromBuildInfo(manifestName, svcName string, b *model.BuildInfo, o *type
...
+			o.Secrets = make([]string, 0)
```
Comment (AgustinRamiroDiaz) [inline pkg/cmd/build/build.go:276]: `o.Secrets = make([]string, 0, len(b.Secrets))`

File: pkg/cmd/build/build.go [hunk 4]
```diff
@@ -271,6 +271,16 @@ func OptsFromBuildInfo(manifestName, svcName string, b *model.BuildInfo, o *type
...
+		}
+		// add to the build the secrets at the manifest
+		for id, src := range b.Secrets {
+			o.Secrets = append(o.Secrets, fmt.Sprintf("id=%s,src=%s", id, src))
```
Comment (AgustinRamiroDiaz) [inline pkg/cmd/build/build.go:280]: Not in the scope for this pr: did you find hard to add this secret with this particular formatting? Should we refactor the `Secrets` to be an array of `struct{id string, src string}`?

**Review by AgustinRamiroDiaz — APPROVED** [review]

File: pkg/cmd/build/build.go (hunk 4 above)
Comment (teresaromero) [reply pkg/cmd/build/build.go:280]: `Secrets` is an option for the `okteto build` command, which is already implemented. This PR enables that this option can be added to the services declared inside the build section of the manifest as the issue describes

File: pkg/model/serializer.go (hunk 1 above)
Comment (teresaromero) [reply pkg/model/serializer.go:49]: i think this is self-explanatory taking into account this is a private struct used under the Unmarshall functions

**Comment by jLopezbarb** [comment]
Can we also translate the compose files that has secrets to the okteto manifest using those?

File: pkg/cmd/build/build.go (hunk 3 above)
Comment (teresaromero) [reply pkg/cmd/build/build.go]: I will go forward with an empty slice with 0 length so it will increment when reading the struct key and value

**Comment by teresaromero** [comment]
> Do we currently support this secrets on the docker compose manifest?

I don't see this is currently implemented under `stack` manifest, for this we should implment it as the spec of compose indicates https://github.com/compose-spec/compose-spec/pull/238/files 

should this be scope of the PR ? 
cc: @jLopezbarb @pchico83

**Review by AdrianPedriza — APPROVED** [review]

**Review by AdrianPedriza — APPROVED** [review]

**Comment by pchico83** [comment]
> > Do we currently support this secrets on the docker compose manifest?
> 
> I don't see this is currently implemented under `stack` manifest, for this we should implment it as the spec of compose indicates https://github.com/compose-spec/compose-spec/pull/238/files
> 
> should this be scope of the PR ? cc: @jLopezbarb @pchico83

@teresaromero let's do it just for the okteto manifest `build` section, that unblocks the scenario

**Review by jLopezbarb — APPROVED** [review]

**Pull request merged**